GetStream/stream-chat-swift
# [CIS-1742] Save channel config as DTO instead of binary data

### 🔗 Issue Link
CIS-1742

### 🎯 Goal
Change how we save the channel config to avoid crashes in schema changes.

### 🛠 Implementation
Instead of storing it as binary data in `ChannelDTO`, we create a new `ChannelConfigDTO`. Storing as binary data is dangerous, because if the structure changes, CoreData doesn't know the schema has changed, so it doesn't do migration or wipe the database.

### 🎨 UI Changes
N/A

### 🧪 Testing

1. Uninstall DemoApp
2. git checkout 73d32a60992b237383cb4f83579720b292123fff
3. Open Han Solo user
4. Close the app
5. git checkout develop
6. Open Han Solo User
7. Should not crash

### ☑️ Contributor Checklist
